glass/observations.py

```diff
@@ -123,6 +123,57 @@ def vmap_galactic_ecliptic(nside, galactic=(30, 90), ecliptic=(20, 80)):
     return m
 
 
+def smail_nz(z, z_mode, alpha, beta):
+    r'''Redshifts following the Smail et al. (1994) model.
+
+    The redshift follows the Smail et al. [1]_ redshift distribution.
+
+    Parameters
+    ----------
+    z : array_like
+        Redshift values of the distribution.
+    z_mode : float or array_like
+        Mode of the redshift distribution, must be positive.
+    alpha : float or array_like
+        Power law exponent (z/z0)^\alpha, must be positive.
+    beta : float or array_like
+        Log-power law exponent exp[-(z/z0)^\beta], must be positive.
+
+    Returns
+    -------
+    pz : array_like
+        Redshift distribution at the given ``z`` values.
+
+    Notes
+    -----
+    The probability distribution function :math:`p(z)` for redshift :math:`z`
+    is given by Amara & Refregier [2]_ as
+
+    .. math::
+
+        p(z) \sim \left(\frac{z}{z_0}\right)^\alpha
+                    \exp\left[-\left(\frac{z}{z_0}\right)^\beta\right] \;,
+
+    where :math:`z_0` is matched to the given mode of the distribution.
+
+    References
+    ----------
+    .. [1] Smail I., Ellis R. S., Fitchett M. J., 1994, MNRAS, 270, 245
+    .. [2] Amara A., Refregier A., 2007, MNRAS, 381, 1018
+
+    '''
+    z_mode = np.asanyarray(z_mode)[..., np.newaxis]
+    alpha = np.asanyarray(alpha)[..., np.newaxis]
+    beta = np.asanyarray(beta)[..., np.newaxis]
+
+    # pz = np.power(z, alpha[..., np.newaxis])
+    # pz *= np.exp((-1) * np.power(z/z0[..., np.newaxis], beta[..., np.newaxis]))
+    pz = z**alpha*np.exp(-alpha/beta*(z/z_mode)**beta)
+    pz /= np.trapz(pz, z, axis=-1)[..., np.newaxis]
```

Comment:
no commented code

```suggestion
    pz = z**alpha*np.exp(-alpha/beta*(z/z_mode)**beta)
    pz /= np.trapz(pz, z, axis=-1)[..., np.newaxis]
```

Reply:
Oh my mistake! I thought I deleted this before staging it!